docs/build.md

```diff
@@ -2,40 +2,88 @@
 
 Dependency | Version | Required           | Runtime
 ---        | ---     | ---                | ---
-Kokkos     | 3.1.00  | :heavy_check_mark: | :heavy_check_mark:
+[Kokkos](https://github.com/kokkos/kokkos)     | 3.1.00  | :heavy_check_mark: | :heavy_check_mark:
 MPI        | 2       |                    | :heavy_check_mark:
 CMake      | 3.12    | :heavy_check_mark: |
 
-# Installation
+# Build and installation
 
-## Installation using Spack
+The Kokkos dependency build instructions can be found [here](https://github.com/kokkos/kokkos#building-and-installing-kokkos).
+ArborX requires Kokkos to be built using C++14 standard
+(`-DKokkos_CXX_STANDARD=14`). In addition, if CUDA backend is desired, Kokkos
+must experimental lambda option enabled (`-DKokkos_ENABLE_CUDA_LAMBDA=ON`).
```

Comment:
```suggestion
must be built with support for `__host__ __device__` lambdas (`-DKokkos_ENABLE_CUDA_LAMBDA=ON`).
```

Reply:
Not sure if we want to be this specific.

Reply:
OK, then
```suggestion
must have the experimental lambda option enabled (`-DKokkos_ENABLE_CUDA_LAMBDA=ON`).
```
(even if that sounds even more obscure to me).

Reply:
Is it still experimental? Or should we just say `must have support for CUDA lambda`?

Reply:
`CUDA` now also allows `--extended-lambda` instead of `--expt-extended-lambda`. It's not listed as experimental anymore. I am also fine just saying "In addition, if the CUDA backend is desired, Kokkos be configured with `-DKokkos_ENABLE_CUDA_LAMBDA=ON`" without any explanations what it's for.